### DISCUSSION TITLE
Foundry101, Raffle, section 34, testCheckUpkeepReturnsFalseIfRaffleIsNotOpen fails

### DISCUSSION THREAD

**Discussion by BowtiedHarpyEagle** [Cyfrin/foundry-full-course-cu#2653] [Q&A] [answered]
I feel totally lost at with this test. It doesn't make sense to me. First of all, under the conditions we are testing, when our prank user enters the raffle, pays enough money, enough time is passed, block number increased, shouldn't the upkeepNeeded be true? And how come the same code in the video shows a passed test when Patrick runs the test, but my test doesn't pass? Also, different code in the video lesson vs text lesson, when I check the text lesson for clarity,  instead I'm more confused. I'll add my relevant test, a screenshot and link to my github repo for the full code.
relevant test:
```sol
 function testCheckUpkeepReturnsFalseIfRaffleIsNotOpen() public {
        //Arrange
        vm.prank(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
        vm.warp(block.timestamp + interval + 1);
        vm.roll(block.number + 1);
        //Act
        (bool upkeepNeeded, ) = raffle.checkUpkeep("");
        //Assert
        assert(!upkeepNeeded);
    }
```
![screen4](https://github.com/user-attachments/assets/b6e9efb8-e0dc-4582-b36b-a6cd4b6d8511)
https://github.com/BowtiedHarpyEagle/cu-ff-s4-lottery/tree/af23b1a51d5bc4f7e756eb30826bddeac58b0231

**Comment by EngrPips** [comment, marked as answer]
Hello @BowtiedHarpyEagle, Below is my test which passes. You should compare with yours and see where you get it wrong 

```js
 function testCheckUpKeepReturnsFalseIfLotteryIsNotOpen() public {
        vm.prank(player);
        lottery.enter{value: _entryFee}();
        vm.warp(block.timestamp + _interval + 1);
        vm.roll(block.number + 1);

        lottery.performUpkeep("0x0");

        ( bool upKeepNeeded,) = lottery.checkUpkeep("0x0");

        assertEq(upKeepNeeded, false);
    }
```

**Reply by BowtiedHarpyEagle** [reply]
I copy-pasted code from the text lesson, adjusted the variable and state names to mine, it is now passing. 

Updated code for the test: 

```solidity
    function testCheckUpkeepReturnsFalseIfRaffleIsntOpen() public {
        // Arrange
        vm.prank(PLAYER);
        raffle.enterRaffle{value: entranceFee}();
        vm.warp(block.timestamp + interval + 1);
        vm.roll(block.number + 1);
        raffle.performUpkeep("");
        Raffle.RaffleState raffleState = raffle.getState();
        // Act
        (bool upkeepNeeded,) = raffle.checkUpkeep("");
        // Assert
        assert(raffleState == Raffle.RaffleState.CALCULATING_WINNER);
        assert(upkeepNeeded == false);
    }
```

**Reply by EngrPips** [reply]
You are a giga chad sir

**Reply by BowtiedHarpyEagle** [reply]
Thanks for that as well. I feel more like an ID10T when I have these issues, but sometimes all I need to do is to get a good nights' sleep, and come back with fresh eyes.

**Reply by EngrPips** [reply]
Exactly, we use to underestimate the power of good rest.

**Comment by EngrPips** [comment]
Hello @BowtiedHarpyEagle, I don't mean to say I need you to mark my response as an answer or whatever, but I want to confirm if there is any reason you unmark the answer. Is there still some issues that haven't been resolved?

**Reply by BowtiedHarpyEagle** [reply]
I did have some issues with this yesterday, but then I realized where I made a mistake.

**Reply by EngrPips** [reply]
Alright then. Keep crushing it.

**Reply by BowtiedHarpyEagle** [reply]
Thanks. You're the ultra mega giga chad, always on the ball, always helpful
